src/runtime_src/xdp/profile/device/device_intf.cpp

```diff
@@ -419,9 +420,21 @@ DeviceIntf::~DeviceIntf()
     char buffer[65536];
     // debug_ip_layout max size is 65536
     ifs.read(buffer, 65536);
+
+
     debug_ip_layout *map;
     if (ifs.gcount() > 0) {
       map = (debug_ip_layout*)(buffer);
+#else
+    size_t sz1 = 0, sectionSz = 0;
+    // Get the size of full debug_ip_layout
+    mDevice->getDebugIpLayout(nullptr, sz1, &sectionSz);
+    // Allocate buffer to retrieve debug_ip_layout information from loaded xclbin
+    char* buffer = new char[sectionSz];
```

Comment:
```
std::vector<char> buffer(sectionSz);
mDevice->getDeviceIPLayout(buffer.data(), sectionSz, &sz1);
auto map = reinterpret_cast<const debug_ip_layout*>(buffer.data());
```

Reply:
I have added the change. Thanks for the suggestion!